Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -488,29 +517,47 @@ int PcapLiveDevice::startCaptureBlockingMode(OnPacketArrivesStopBlocking onPacke
 	m_cbOnPacketArrivesBlockingMode = onPacketArrives;
 	m_cbOnPacketArrivesBlockingModeUserCookie = userCookie;
 
-	long startTimeSec = 0, startTimeNSec = 0;
-	clockGetTime(startTimeSec, startTimeNSec);
-
-	long curTimeSec = 0;
+	auto startTime = std::chrono::steady_clock::now();
 
+	auto currentTime = startTime;
 	m_CaptureThreadStarted = true;
 	m_StopThread = false;
 
-	if (timeout <= 0)
+#if !defined(_WIN32)
+	struct pollfd pcapPollFd;
+	memset(&pcapPollFd, 0, sizeof(pcapPollFd));
+	pcapPollFd.fd = m_PcapSelectableFd;
+	pcapPollFd.events = POLLIN;
+	constexpr int pollShortIntervalMs = 10;
+#endif
+
+	while (!m_StopThread)
 	{
-		while (!m_StopThread)
+		if(m_usePoll)
 		{
-			pcap_dispatch(m_PcapDescriptor, -1, onPacketArrivesBlockingMode, (uint8_t*)this);
```

Comment:
notice here `onPacketArrivesBlockingMode` is renamed to `onPacketArrivesAndStopBlockingMode`